nexus/auth/src/context.rs

```diff
@@ -346,19 +353,42 @@ mod test {
     use crate::authn;
     use crate::authz;
     use authz::Action;
-    use nexus_test_utils::db::test_setup_database;
+    use nexus_db_model::IdentityType;
+    use nexus_db_model::RoleAssignment;
     use omicron_common::api::external::Error;
+    use omicron_common::api::external::ResourceType;
     use omicron_test_utils::dev;
     use std::collections::BTreeMap;
     use std::sync::Arc;
+    use uuid::Uuid;
+
+    struct FakeStorage {}
+
+    impl FakeStorage {
+        fn new() -> Arc<dyn crate::storage::Storage> {
+            Arc::new(Self {})
+        }
+    }
+
+    #[async_trait::async_trait]
+    impl crate::storage::Storage for FakeStorage {
+        async fn role_asgn_list_for(
+            &self,
+            _opctx: &OpContext,
+            _identity_type: IdentityType,
+            _identity_id: Uuid,
+            _resource_type: ResourceType,
+            _resource_id: Uuid,
+        ) -> Result<Vec<RoleAssignment>, Error> {
+            todo!();
+        }
+    }
```

Comment:
Another example of a test that previously required a whole database, but actually, never queries anything from the database